src/routes/v1/friendships/history.rs

```diff
@@ -0,0 +1,101 @@
+use actix_http::HttpMessage;
+use actix_web::{
+    get,
+    web::{self, Data},
+    HttpRequest, HttpResponse,
+};
+use chrono::NaiveDateTime;
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+use crate::{
+    components::app::AppComponents,
+    entities::friendship_history::FriendshipHistory,
+    middlewares::check_auth::UserId,
+    routes::v1::{
+        error::CommonError,
+        friendships::{errors::FriendshipsError, types::MessageRequestEventResponse},
+    },
+};
+
+use super::types::MessageRequestEvent;
+
+#[derive(Deserialize, Serialize)]
+pub struct RequestEventRequestBody {
+    pub timestamp_from: i64, // timestamp in milis
+    pub timestamp_to: i64,   // timestamp in milis
+}
+
+#[get("/v1/friendships/{friendshipId}/request-events/messages")]
+async fn get_sent_messages_request_event(
+    req: HttpRequest,
+    body: web::Json<RequestEventRequestBody>,
+    friendship_id: web::Path<String>,
+    app_data: Data<AppComponents>,
+) -> Result<HttpResponse, FriendshipsError> {
+    let _logged_in_user = req
+        .extensions()
+        .get::<UserId>()
+        .expect("to have a UserId")
+        .clone();
+
+    // Convert it to a timestamp type that can be understood by PostgreSQL.
+    let timestamp_from_naive: NaiveDateTime =
+        NaiveDateTime::from_timestamp_opt(body.timestamp_from, 0).unwrap();
+    let timestamp_to_naive: NaiveDateTime =
+        NaiveDateTime::from_timestamp_opt(body.timestamp_to, 0).unwrap();
+
+    // Get the history of friendship request events.
+    match &app_data.db.db_repos {
+        Some(repos) => {
+            let (history, _) = repos
+                .friendship_history
+                .get_friendship_request_event_history(
+                    friendship_id.parse::<Uuid>().unwrap(),
```

Comment:
It would be great if we handle the error here too because the `firendship_id` is something that the client sends within the payload. If it sends a 'aaaa', it's not a valid `Uuid` . Maybe you can do this (not totally sure if it works  for the `Uuid` type but it could) : 

```rust
async fn get_sent_messages_request_event(
    req: HttpRequest,
    body: web::Json<RequestEventRequestBody>,
    friendship_id: web::Path<Uuid>, // If Actix cannot convert it to Uuid, it will automatically sends back a 400 Bad Request
    app_data: Data<AppComponents>,
) 
```

Let's check if the code above works.

Reply:
Worked perfectly. I just had to add this [line](https://github.com/decentraland/social-service/pull/122/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542R31) to enable the `serde` feature flag on the `uuid` crate in order for it to provide `Serialize` and `Deserialize` impls.